sqldelight-compiler/src/main/kotlin/app/cash/sqldelight/core/lang/queryType.kt

```diff
@@ -0,0 +1,20 @@
+package app.cash.sqldelight.core.lang
+
+import com.squareup.kotlinpoet.ClassName
+import com.squareup.kotlinpoet.ParameterizedTypeName.Companion.parameterizedBy
+import com.squareup.kotlinpoet.TypeName
+
+private val QUERY_TYPE = ClassName("app.cash.sqldelight", "Query")
+
+/**
+ * Parameterize a `Query` by two types: [dataType] and [cursorType].
+ */
+fun parameterizeQueryBy(dataType: TypeName, cursorType: TypeName) = QUERY_TYPE.parameterizedBy(
+  dataType, cursorType
+)
+
+const val QUERIES_SUFFIX_NAME = "Queries"
+internal val QUERY_FUNCTION = ClassName("app.cash.sqldelight", "Query")
```

Comment:
I'm not very familiar with KotlinPoet. It seems weird that we were using `QUERY_TYPE` to import a query builder function. If there's a better way to do this that doesn't use `ClassName` let me know!

Reply:
The original behavior can be seen here! https://github.com/cashapp/sqldelight/pull/2827/files#diff-075e4b633563d5d4193593fafa8944ef2fbfe9a665d3c408fc7c9361956e183eR219